lib/ansible/plugins/loader.py

```diff
@@ -208,7 +209,9 @@ def _load_config_defs(self, name, path):
 
             # if type name != 'module_doc_fragment':
             if type_name in C.CONFIGURABLE_PLUGINS:
-                dstring = get_docstring(path, fragment_loader, verbose=False, ignore_errors=True)[0]
+                dstring = AnsibleLoader(getattr(module, 'DOCUMENTATION', ''), file_name=path).get_single_data()
+                if dstring:
+                    add_fragments(dstring, path, fragment_loader=fragment_loader)
```

Comment:
We have functions to take care of this.

-1

Reply:
not currently, you describe a new api you want to build in previous comment, this is just quick change to improve performance until new API is in place

Reply:
No... Actually I describe an API that you told me you were going to implement after the config framework was implemented.  Since you didn't implement in then, it's time for you to implement it now.

Reply:
I'm unsure what API you are talking about, the config framework was basically the API to be implemented. 

I had no plans on doc reading API. This PR was mostly inspiration that plugins that support config are always importable and hence we can access w/o AST parsing as modules are not guaranteed the same.